src/attributes/type_system.md

```diff
@@ -163,12 +163,33 @@ match message {
 }
 ```
 
-It's also not allowed to cast non-exhaustive types from foreign crates.
+However, casting non-exhaustive types from foreign crates is generally disallowed, except when dealing with enums that have no non-exhaustive variants.
```

Comment:
I'm not a maintainer of the reference, so this will need a review from someone else as well.

I'm not sure if casting any other type other than an enum is ever allowed in Rust. I can't think of any, and if there are some, it might be good to mention them.

A quick suggestion to avoid starting with `However` in two nearby paragraphs:
```suggestion
It's also not allowed to use numeric casts (`as`) on enums that contain any non-exhaustive variants.
```

Reply:
whoops completely missed the duplication of "However" 😬. Yeah I don't think anything other than an enum can be cast as well. I'll commit the suggestion.